dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowFactoryImpl.java

```diff
@@ -485,43 +487,66 @@ public WorkflowStep findStep(String id) throws DotDataException {
 		return step;
 	}
 
-	public WorkflowStep findStepByContentlet(Contentlet contentlet) throws DotDataException {
-		WorkflowStep step = cache.getStep(contentlet);
-		final WorkflowScheme scheme = this.findSchemeForStruct(contentlet.getStructureInode());
-		if ((step == null) || !step.getSchemeId().equals(scheme.getId())) {
+	public List<WorkflowStep> findStepsByContentlet(Contentlet contentlet) throws DotDataException {
+		List<WorkflowStep> steps = new ArrayList<>();
+        List<WorkflowStep> currentSteps = cache.getSteps(contentlet);
+		final List<WorkflowScheme> schemes = this.findSchemeForStruct(contentlet.getContentTypeId());
+		if (currentSteps == null) {
+            WorkflowStep step = null;
 			try {
 				final DotConnect db = new DotConnect();
 				db.setSQL(sql.SELECT_STEP_BY_CONTENTLET);
 				db.addParam(contentlet.getIdentifier());
-				step = (WorkflowStep) this.convertListToObjects(db.loadObjectResults(), WorkflowStep.class).get(0);
+                step = (WorkflowStep) this.convertListToObjects(db.loadObjectResults(), WorkflowStep.class).get(0);
+                steps.add(step);
+
 			} catch (final Exception e) {
 				Logger.debug(this.getClass(), e.getMessage());
 			}
 
 			if (step == null) {
 				try {
-					step = this.findSteps(scheme).get(0);
+					for(WorkflowScheme scheme : schemes) {
+						step = this.findSteps(scheme).get(0);
+						steps.add(step);
+					}
+					//Add to cache list of steps
 				} catch (final Exception e) {
 					throw new DotDataException("Unable to find workflow step for content id:" + contentlet.getIdentifier());
 				}
 			}
 
-			// if the existing task belongs to another workflow schema, update
-			// to the latest schema
-			if (!step.getSchemeId().equals(scheme.getId())) {
-				step = this.findSteps(scheme).get(0);
-				final DotConnect db = new DotConnect();
-				db.setSQL(sql.RESET_CONTENTLET_STEPS);
-				db.addParam(step.getId());
-				db.addParam(contentlet.getIdentifier());
-				db.loadResult();
 
-			}
-			cache.addStep(contentlet, step);
+		}else {
+			steps.addAll(currentSteps);
 		}
-		return step;
+        // if the existing task belongs to another workflow schema, then blank
+        // the workflow task status
+		if (steps != null  && steps.size() == 1 && !existSchemeIdOnSchemesList(steps.get(0).getSchemeId(),schemes)) {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)